airbyte-integrations/connectors/source-stripe/source_stripe/schemas/invoices.json

```diff
@@ -110,7 +110,7 @@
       "description": "Any discounts applied to the invoice",
       "type": ["null", "array"],
       "items": {
-        "type": ["null", "string"]
+        "type": ["null", "object"]
```

Comment:
Hi @marcosmarxm - here's the link to [invoice.discounts now being an array of objects] when expandable (https://docs.stripe.com/api/invoices/object#invoice_object-discounts) and [how expandables work in Stripe](https://docs.stripe.com/api/expanding_objects)- and the [reference to Stripe's discount object](https://docs.stripe.com/api/discounts)

Reply:
The [docs](https://docs.stripe.com/api/invoices/object#invoice_object-discounts) says it should be the `array of strings`? 
Reading docs more resulted in the fact the `discounts` should be the `array of objects`, with the following structure: https://docs.stripe.com/api/discounts/object

This means we need to declare this in the schema as well. The changes are needed or the explanation of why we leave the `object` only and don't expand the schema otherwise.

Thank you.

Reply:
Two concerns:
* Using expandable, I still get an array of string. Yes, they look a lot like strings but if Stripe consider them as string in their API doc, I wouldn't mess with that as they could be sending strings that do not represent objects. Trace from our test account for documentation purposes:
```
            "discounts": [
                "{'id': 'di_1PDuEkEcXtiJtvvhSoPOEqOo', 'object': 'discount', 'checkout_session': None, 'coupon': {'id': 'iJ6qlwM5', 'object': 'coupon', 'amount_off': None, 'created': 1674208993, 'currency': None, 'duration': 'forever', 'duration_in_months': None, 'livemode': False, 'max_redemptions': None, 'metadata': {}, 'name': '\u0415\u0443\u0456\u0435', 'percent_off': 10.0, 'redeem_by': None, 'times_redeemed': 5, 'valid': True}, 'customer': 'cus_NGoTFiJFVbSsvZ', 'end': None, 'invoice': 'in_1PDuE1EcXtiJtvvhwv4F8wEK', 'invoice_item': None, 'promotion_code': None, 'start': 1715112222, 'subscription': None, 'subscription_item': None}"
            ],
```
* A little bit of background on how incremental syncs work: Stripe does not allow to filter on the resources endpoints using "updated" as a parameter. Instead, they provide another endpoint which is the events endpoint. That being said: I'm very confused about Stripe behavior regarding this field as poking the resource directly, we get the snippet I share in the previous point. However, when querying the events endpoints, I get the following:
```
          "discounts": [
            "di_1PDuEkEcXtiJtvvhSoPOEqOo"
          ],
```
I can't explain the discrepancy but would like you to confirm this behavior on your side: Do you have the same discounts values in full_refresh and incremental? I'll contact Stripe regarding this but they already told us that `events [...] provide a summary of the object's data at the time of the event, without including any expanded properties.` so I don't expect them to support those. It seems like the pattern we use for `charges` surprisingly so I'll try to get more information from them as to why

Reply:
Strange behaviour from Stripe - unsure why this happens but I'll revert the schema back - I only noticed it now that I looked at the output again.